langroid/agent/base.py

```diff
@@ -322,6 +323,52 @@ def _get_tool_list(
                 lambda msg: message_class.handle_message_fallback(self, msg),
             )
 
+        async_tool_name = f"{tool}_async"
+        if (
+            hasattr(message_class, "handle_async")
+            and inspect.isfunction(message_class.handle_async)
+            and not hasattr(self, async_tool_name)
+        ):
+            has_chat_doc_arg = (
+                len(inspect.signature(message_class.handle_async).parameters) > 2
```

Comment:
shouldn't this be > 1 just like in the "handle" case?

Reply:
@pchalasani I think you are right, but it's not something that I've changed; this line was modified on 2028/08/11 by you as part of #539.

Reply:
Right, there are two cases of `has_doc_chat_arg`, one for `handle` and one for `response`, the first one checks for > 1 param and the second one checks for > 2 params. In the **async** version you added, you're using > 2 in both cases. I'll push my change and merge. Thanks @alexagr

Reply:
Yes, you're right, thanks for catching this!